src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -927,11 +927,12 @@ private enum PendingOperations
                 break;
             } while (true);
 
-            var status = GetPendingOperationsStatus(context, pendingOps.Count == 0);
+            var currentOperationsCount = _operations.Count;
+            var status = GetPendingOperationsStatus(context, currentOperationsCount == 0);
```

Comment:
It was 'pendingOps' which was renamed to 'executedOps'. Now you are using '_operations.Count'. Is this correct?

Reply:
Yes, `pendingOps` were actually the executed ops (see line 870 we get it out of `_operations` and execute on line 897), then we check for more work using `pendingOps ` (line 930) which will always return `HasMore` and we will make another run of transactionmerger. Also see the logging on line 934 it was actually counting the remaining ops using the _operation list.